src/webgl/primitives.js

```diff
@@ -151,13 +151,18 @@ p5.prototype.box = function(){
           this.uvs.push([j & 1, (j & 2) / 2]);
           id++;
         }
-        this.faces.push([v, v + 1, v + 2]);
-        this.faces.push([v + 2, v + 1, v + 3]);
+        /**THIS IS HOW FACES WERE ORIGINALLY CALCULATED FOR BOX***/
+        /**TEMPORARILY DOING FACE CALCULATION IN _edgesToVertices**/
+        // this.faces.push([v, v + 1, v + 2]);
+        // this.faces.push([v + 2, v + 1, v + 3]);
       }
     };
     var boxGeom = new p5.Geometry(detailX,detailY, _box);
     boxGeom
-      .computeNormals();
+      //.computeNormals(); NOT NECESSARY CURRENTLY
+      ._makeTriangleEdges()
+      ._edgesToVertices();
```

Comment:
I think it would make sense to just store it in one object and send different values to the shader depending on which parts you want to draw but maybe they would need to be different geometries. Not sure yet but definitely worth considering.